### PR TITLE
fc: integrate Firecracker's metrics

### PR DESCRIPTION
**Pull request by liubin**
Firecracker expose metrics through fifo file and using a JSON format.
This PR will parse the Firecracker's metrics and convert to Prometheus metrics.

Fixes: #472

Signed-off-by: bin liu <bin@hyper.sh>

### CODE REVIEW

File: src/runtime/virtcontainers/fc_metrics.go [hunk 1]
```diff
@@ -0,0 +1,761 @@
...
+
+package virtcontainers
+
+import (
+	"encoding/json"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const namespaceFirecracker = "kata_firecracker"
```
Comment (jodh-intel) [inline src/runtime/virtcontainers/fc_metrics.go:14]: Nit: You could shorten this to something like `fcMetricsNS`.

File: src/runtime/virtcontainers/fc_metrics.go [hunk 2]
```diff
@@ -0,0 +1,761 @@
...
+	var fm FirecrackerMetrics
+	if err := json.Unmarshal([]byte(line), &fm); err != nil {
+		fc.Logger().WithError(err).WithField("data", line).Error("failed to unmarshal fc metrics")
+		return
+	}
+	updateFirecrackerMetrics(&fm)
+}
+
+//
+// below is codes generated from
```
Comment (jodh-intel) [inline src/runtime/virtcontainers/fc_metrics.go:26]: Could you put this auto-generated code in a different file maybe with a warning comment at the top like https://github.com/kata-containers/kata-containers/blob/2.0-dev/src/runtime/cli/config-generated.go.in:

```golang
// WARNING: This file is auto-generated - DO NOT EDIT!
```

Comment (liubin) [reply src/runtime/virtcontainers/fc_metrics.go:26]: Yes, it's better. I can put other codes into fc.go, and leave this as auto-generated.
